xla/pjrt/transpose.h

```diff
@@ -240,7 +240,18 @@ class TransposePlan {
   int64_t scratch_size_ = 0;
 };
 
-struct TransposePlanCacheKey;
+struct TransposePlanCacheKey {
+  size_t elem_size_in_bytes;
+  absl::InlinedVector<int64_t, 4> dims;
+  absl::InlinedVector<int64_t, 4> permutation;
+  bool input_layout_is_tiling;
+  absl::InlinedVector<int64_t, 4> input_layout;
+  absl::InlinedVector<int64_t, 4> output_tiling;
+  TransposePlan::Transformation transformation;
+  int num_threads;
+
+  bool operator==(const TransposePlanCacheKey& other) const;
+};
 
 template <typename H>
 H AbslHashValue(H h, const TransposePlanCacheKey& key);
```

Comment:
It's not your fault, but can you see about moving this into being a private member of TransposePlanCache (the hash function might have to be passed to the LRUCache then)? It's not great that this is only used inside the class but is being made public to more than needed.

Reply:
From simplicity perspective, the ugliness is even worse now 🤣. But whatever, lets do it.